internal/cli/login.go

```diff
@@ -14,34 +14,39 @@ import (
 
 var (
 	loginTenantDomain = Flag{
-		Name:         "Tenant Domain",
-		LongForm:     "domain",
-		Help:         "Specifies tenant domain when authenticating via client credentials (client ID, client secret)",
-		IsRequired:   false,
-		AlwaysPrompt: true,
+		Name:       "Tenant Domain",
+		LongForm:   "domain",
+		Help:       "Specifies tenant domain when authenticating via client credentials (client ID, client secret)",
+		IsRequired: false,
 	}
 
 	loginClientID = Flag{
-		Name:         "Client ID",
-		LongForm:     "client-id",
-		Help:         "Client ID of the application.",
-		IsRequired:   true,
-		AlwaysPrompt: true,
+		Name:       "Client ID",
+		LongForm:   "client-id",
+		Help:       "Client ID of the application.",
+		IsRequired: false,
 	}
 
 	loginClientSecret = Flag{
-		Name:         "Client Secret",
-		LongForm:     "client-secret",
-		Help:         "Client Secret of the application.",
-		IsRequired:   true,
-		AlwaysPrompt: true,
+		Name:       "Client Secret",
+		LongForm:   "client-secret",
+		Help:       "Client Secret of the application.",
+		IsRequired: false,
+	}
+
+	loginAsUser = Flag{
+		Name:       "Login as user",
+		LongForm:   "as-user",
+		Help:       "Initializes login as a user via device code flow.",
+		IsRequired: false,
```

Comment:
I know that we removed the `--as-machine` flag previously, but I did not know of another way of skipping the interactive prompt and proceeding straight to the user login mechanism. 

Alternatively, if this is not a use case that we wish to satisfy, we can always remove and force the user to interact with the initial prompt. But my hunch is that more inclined users would want a way to skip that. It may also help with certain quickstarts and tutorials.

Reply:
**Update:** @sergiught and I decided to nix the `--as-user` flag in favor of the `--no-input` flag because it already exists and satisfies that functionality requirement. Further, we don't expect users to be needing the "skip" mechanism. very often, so it is ok to slightly deprioritize.